### PR TITLE
Add ABS OP

### PR DESCRIPTION
**Pull request by yajiedesign**
add abs op

### CODE REVIEW

**Comment by piiswrong** [comment]
@tqchen This probably should go into activations. What do you think?

**Comment by tqchen** [comment]
Add unary operations here https://github.com/dmlc/mxnet/blob/master/src/ndarray/unary_function-inl.h#L84
 
Which should be easier than creating an op from scratch

**Comment by tqchen** [comment]
@yajiedesign sorry that we did not make things clear beforehand, can you change your implementation to use what I pointed out instead?

**Comment by yajiedesign** [comment]
@tqchen i can change,but i am test like this 
data = mx.symbol.Variable('data')
test = mx.nd.square(data)
it not work.What should I do?
error info is 
Traceback (most recent call last):
  File "test_mlp2.py", line 44, in <module>
    test_mlp()
  File "test_mlp2.py", line 26, in test_mlp
    test = mx.nd.square(data)
  File "E:\pyjq\cpython\lib\site-packages\mxnet-0.5.0-py3.6.egg\mxnet\ndarray.py", line 638, in unary_ndarray_function
    c_array(NDArrayHandle, (out.handle,))))
OSError: exception: access violation reading 0xFFFFFFFFFFFFFFF

**Comment by tqchen** [comment]
You need to use ```mx.sym.square``` for raking symbolic squares

File: src/ndarray/unary_function-inl.h [hunk 1]
```diff
@@ -84,6 +84,10 @@ void L2Norm(const TBlob &src,
 }
 // Register all unary operations here
 // The true means inplace can be enabled.
+MXNET_REGISTER_TBLOB_FUN(abs, XPU)
+.set_function(XPU::kDevMask, UnaryForward_<XPU, op::mshadow_op::abs>, true)
+.set_gradient(XPU::kDevMask, UnaryBackwardUseIn_<XPU, op::mshadow_op::abs_grad>, true)
```
Comment (tqchen) [inline src/ndarray/unary_function-inl.h]: change it to mshadow_op::sign

File: src/operator/mshadow_op.h [hunk 2]
```diff
@@ -117,6 +117,26 @@ struct threshold {
...
+/*! \brief used for generate element of abs */
+struct abs {
+  MSHADOW_XINLINE static real_t Map(real_t a) {
+    return fabs(a);
+  }
+};
+
+struct abs_grad {
+  MSHADOW_XINLINE static real_t Map(real_t a) {
+    return (real_t(0) < a) - (a < real_t(0));
```
Comment (tqchen) [inline src/operator/mshadow_op.h]: if (a < 0.0f) return -1.0f;
if (a > 0.0f) return 1.0f;
return 0.0f

File: src/operator/mshadow_op.h [hunk 3]
```diff
@@ -117,6 +117,26 @@ struct threshold {
   }
 };
 
+/*! \brief used for generate element of abs */
+struct abs {
+  MSHADOW_XINLINE static real_t Map(real_t a) {
+    return fabs(a);
```
Comment (tqchen) [inline src/operator/mshadow_op.h]: fabsf

File: src/ndarray/unary_function-inl.h [hunk 4]
```diff
@@ -84,6 +84,10 @@ void L2Norm(const TBlob &src,
...
+.describe("Take square of the src");
```
Comment (tqchen) [inline src/ndarray/unary_function-inl.h]: Take absolute value of the src

File: src/operator/mshadow_op.h [hunk 5]
```diff
@@ -117,6 +117,26 @@ struct threshold {
...
+  }
+};
+
+struct abs_grad {
```
Comment (tqchen) [inline src/operator/mshadow_op.h]: delete abs_grad, since this is sign

**Comment by tqchen** [comment]
Last few more comments, please fix these, and we can merge it in

**Pull request merged**